.github/workflows/msbuild.yml

```
@@ -79,6 +79,9 @@ jobs:
     - if: runner.os != 'Windows'
       run: |
         unzip dafny/Package/CI.zip -d unzippedRelease
+    - name: Run XUnit tests
+      if: {{ matrix.shard }} == 0
+      run: dotnet test --no-restore --verbosity normal dafny/Source/Dafny.sln
```

```suggestion
      run: dotnet test --no-restore --verbosity normal dafny/Source/DafnyPipeline.Test/DafnyPipeline.Test.csproj
```

Otherwise you're running the `DafnyLanguageServer.Test` tests twice.

In fact, it would be better to rename `language-server.yml` to be just `xunit-tests.yml` or something like that, and test the whole solution there. Then you also don't need the shard hack.

---

Thanks, will do the latter.

---

Great, that's one one fewer thing the xUnit PR has to do anyway :)